src/main/java/org/fcrepo/storage/ocfl/DefaultOcflObjectSession.java

```diff
@@ -581,9 +582,14 @@ private void touchRelatedResources(final ResourceHeaders headers) {
         }
     }
 
+    private static String generateStateToken(final Instant timestamp) {
+        return DigestUtils.md5Hex(String.valueOf(timestamp.toEpochMilli())).toUpperCase();
+    }
+
     private void touchResource(final String resourceId, final Instant timestamp) {
         final var headers = ResourceHeaders.builder(readHeaders(resourceId))
                 .withLastModifiedDate(timestamp)
+                .withStateToken(generateStateToken(timestamp))
```

Comment:
Rather than generating this copy the value from the resource headers of the resource that changed.